### ISSUE TITLE
Error: The terraform-provider-azurerm_v3.30.0_x5 plugin crashed!

### ISSUE THREAD

**Issue by oussama-noble-ai** [hashicorp/terraform-provider-azurerm#20461]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.7

### AzureRM Provider Version

3.30.0

### Affected Resource(s)/Data Source(s)

azurerm_redis_cache, azurerm_storage_account, azurerm_key_vault_secret, azurerm_container_group

### Terraform Configuration Files

```hcl
Terraform v1.3.7
on darwin_arm64
+ provider registry.terraform.io/azure/azapi v1.0.0
+ provider registry.terraform.io/hashicorp/aws v3.59.0
+ provider registry.terraform.io/hashicorp/azuread v2.31.0
+ provider registry.terraform.io/hashicorp/azurerm v3.30.0
+ provider registry.terraform.io/hashicorp/null v3.2.1
+ provider registry.terraform.io/hashicorp/random v3.3.2
```

### Debug Output/Panic Output

```shell
The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵
Releasing state lock. This may take a few moments...

Stack trace from the terraform-provider-azurerm_v3.30.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x1048afd90]

goroutine 838 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/redis.flattenRedisIdentity(0x140025e4ed0)
	github.com/hashicorp/terraform-provider-azurerm/internal/services/redis/redis_cache_resource.go:1081 +0x150
github.com/hashicorp/terraform-provider-azurerm/internal/services/redis.resourceRedisCacheRead(0x140035d8400, {0x105804ca0?, 0x140016b2c00?})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/redis/redis_cache_resource.go:661 +0x560
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x106241240?, {0x106241240?, 0x14001c8f080?}, 0xd?, {0x105804ca0?, 0x140016b2c00?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/resource.go:712 +0x138
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0x14000e29ea0, {0x106241240, 0x14001c8f080}, 0x14001ddadd0, {0x105804ca0, 0x140016b2c00})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/resource.go:1015 +0x46c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0x1400047e5e8, {0x106241240?, 0x14001c8efc0?}, 0x14002bafe40)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/grpc_provider.go:613 +0x40c
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0x1400097db80, {0x106241240?, 0x14001c8eab0?}, 0x14001d1ccc0)
	github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/tf5server/server.go:747 +0x39c
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x105e4ef20?, 0x1400097db80}, {0x106241240, 0x14001c8eab0}, 0x14001d1cc60, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:349 +0x174
google.golang.org/grpc.(*Server).processUnaryRPC(0x140002a6c40, {0x1062534d8, 0x14000b4c680}, 0x14001bffc20, 0x140016475f0, 0x10a3a73d0, 0x0)
	google.golang.org/grpc@v1.47.0/server.go:1283 +0xb64
google.golang.org/grpc.(*Server).handleStream(0x140002a6c40, {0x1062534d8, 0x14000b4c680}, 0x14001bffc20, 0x0)
	google.golang.org/grpc@v1.47.0/server.go:1620 +0x840
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.47.0/server.go:922 +0x88
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.47.0/server.go:920 +0x298

Error: The terraform-provider-azurerm_v3.30.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behaviour

`terraform plan -out=new.plan` and `terraform apply new.plan` should complete successfully  

### Actual Behaviour

terraform-provider-azurerm_v3.30.0_x5 plugin crashes

### Steps to Reproduce

`terraform init`
`terraform plan -out=new.plan`
`terraform apply new.plan`

So far tried:
1. Closing other apps to free up memory
2. Export GODEBUG=asyncpreemptoff=1
3. Deleting my .terraform folder and lock file
4. Upgrading the azurem module to 3.34.0
5. Upgrading terraform to 1.3.8

Still hitting the same issue

**Comment by aristosvo** [comment]
Hi @oussama-noble-ai! Thanks for filing this issue!

You are lucky, this problem is probably resolved by the migration to a new SDK in #20313. If it is not resolved in the upcoming `v3.43.0`, please let us know!

**Comment by oussama-noble-ai** [comment]
Great! can confirm its not crashing when using `v3.43.0`
